File: BitManipulation/190. 颠倒二进制位/190. 颠倒二进制位.py
class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        for i in range(32):
            bit = 1 & (n >> i)
            # 2進制與10進制處理方式不同
            # res = res*10 + bit
            # bit由左往右推(31-i)位
            res += bit << (31 - i)
        
        return res

File: BitManipulation/190. 颠倒二进制位/190. 颠倒二进制位.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;

        for(int i = 0; i < 32; i++){
            // 取最左數
            uint32_t bit = 1 & (n >> i);
            // bit由左往右推(31-i)位
            res += bit << (31 - i);
        }

        return res;
    }
};